activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -732,11 +732,31 @@ def raw_execute(sql, name, async: false, allow_retry: false, uses_transaction: t
           log(sql, name, async: async) do
             with_raw_connection(allow_retry: allow_retry, uses_transaction: uses_transaction) do |conn|
               sync_timezone_changes(conn)
-              conn.query(sql)
+              result = conn.query(sql)
+              handle_warnings(sql)
+              result
             end
           end
         end
 
+        def handle_warnings(sql)
+          return if @raw_connection.warning_count == 0
+
+          result = @raw_connection.query("SHOW WARNINGS")
```

Comment:
This will always execute an extra query even if the configuration is `:ignore`. Could/should we skip this in that case?